integration/tests/api/docker_test.go

```diff
@@ -270,3 +270,22 @@ func TestDockerContainerNetworkStats(t *testing.T) {
 	assert.NotEqual(stat.Network.RxBytes, stat.Network.TxBytes, "Network tx and rx bytes should not be equal")
 	assert.NotEqual(stat.Network.RxPackets, stat.Network.TxPackets, "Network tx and rx packets should not be equal")
 }
+
+// Tests goroutine leaks in docker handler.
+func TestGoRoutineSanity(t *testing.T) {
+	fm := framework.New(t)
+
+	assert := assert.New(t)
+	initialGoRoutines, err := fm.Cadvisor().GetGoRoutines()
+	assert.NoError(err)
+	containerId1 := fm.Docker().RunPause()
+	containerId2 := fm.Docker().RunPause()
+	waitForContainer(containerId1, fm)
+	waitForContainer(containerId2, fm)
+	fm.Cleanup()
```

Comment:
AFAIK Cleanup() is idempotent. It will attempt to cleanup and fail, which
is probably OK. I really don't think that this is an issue with this test
though.

On Wed, Dec 9, 2015 at 1:38 PM, Yu-Ju Hong <notifications@github.com> wrote:

> In integration/tests/api/docker_test.go
> <https://github.com/google/cadvisor/pull/1008#discussion_r47155912>:
>
> > @@ -270,3 +270,22 @@ func TestDockerContainerNetworkStats(t *testing.T) {
> >  	assert.NotEqual(stat.Network.RxBytes, stat.Network.TxBytes, "Network tx and rx bytes should not be equal")
> >  	assert.NotEqual(stat.Network.RxPackets, stat.Network.TxPackets, "Network tx and rx packets should not be equal")
> >  }
> > +
> > +// Tests goroutine leaks in docker handler.
> > +func TestGoRoutineSanity(t *testing.T) {
> > +	fm := framework.New(t)
> > +
> > +	assert := assert.New(t)
> > +	initialGoRoutines, err := fm.Cadvisor().GetGoRoutines()
> > +	assert.NoError(err)
> > +	containerId1 := fm.Docker().RunPause()
> > +	containerId2 := fm.Docker().RunPause()
> > +	waitForContainer(containerId1, fm)
> > +	waitForContainer(containerId2, fm)
> > +	fm.Cleanup()
>
> Is there any undesirable side-effect of calling the Cleanup() function
> twice? If not, it might still be worth adding defer fm.Cleanup() to
> handle misbehaved tests
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/1008/files#r47155912>.
>